docker/pudl_etl.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/bash
+function send_slack_msg() {
+    curl -X POST -H "Content-type: application/json" -H "Authorization: Bearer ${SLACK_TOKEN}" https://slack.com/api/chat.postMessage --data "{\"channel\": \"C03FHB9N0PQ\", \"text\": \"$1\"}"
+}
+
+function authenticate_gcp() {
+    # Set the default gcloud project id so the zenodo-cache bucket
+    # knows what project to bill for egress
+    gcloud config set project $GCLOUD_BILLING_PROJECT
+}
+
+function run_pudl_etl() {
+    send_slack_msg ":large_yellow_circle: Deployment started for $GITHUB_SHA-$GITHUB_REF"
+    authenticate_gcp \
+    && pudl_setup \
+        --pudl_in $CONTAINER_PUDL_IN \
+        --pudl_out $CONTAINER_PUDL_OUT \
+    && ferc1_to_sqlite \
+        --clobber \
+        --gcs-cache-path gs://zenodo-cache.catalyst.coop \
+        --bypass-local-cache \
+        $PUDL_SETTINGS_YML \
+    && pudl_etl \
+        --clobber \
+        --gcs-cache-path gs://zenodo-cache.catalyst.coop \
+        --bypass-local-cache \
+        $PUDL_SETTINGS_YML \
+    && pytest \
+        --gcs-cache-path gs://zenodo-cache.catalyst.coop \
+        --bypass-local-cache \
+        --live-dbs test
+}
+
+function shutdown_vm() {
+    gsutil -m cp -r $CONTAINER_PUDL_OUT "gs://pudl-etl-logs/$GITHUB_SHA-$GITHUB_REF"
+
+    echo "Shutting down VM."
+    # # Shut down the deploy-pudl-vm instance when the etl is done.
+    # ACCESS_TOKEN=`curl \
+    #     "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token" \
+    #     -H "Metadata-Flavor: Google" | jq -r '.access_token'`
+
+    # curl -X POST -H "Content-Length: 0" -H "Authorization: Bearer ${ACCESS_TOKEN}" https://compute.googleapis.com/compute/v1/projects/catalyst-cooperative-pudl/zones/us-central1-a/instances/deploy-pudl-vm/stop
+}
+
+function notify_slack() {
+    # Notify pudl-builds slack channel of deployment status
+    if [ $1 = "success" ]; then
+        message=":large_green_circle: Deployment Succeeded\n\n "
```

Comment:
I know this is silly but can we use some more fun emojis in the notifications? 🦄 🌈  / ☠️ 💣

Reply:
Done!